lib/src/uri_template.dart

```diff
@@ -542,11 +542,11 @@ List _splitLiteral(String literal) {
     _fragmentOrQueryRegex,
     onMatch: (m) {
       subparts.add(m);
-      return null;
+      return '';
```

Comment:
curious about these as well

Reply:
if you look at the context, the return value isn't used. These callbacks are used to populate another data structure so the return value doesn't matter